bgpd/bgp_attr.c

```diff
@@ -2827,10 +2829,18 @@ bgp_packet_mpattr_start (struct stream *s, afi_t afi, safi_t safi, afi_t nh_afi,
 
   stream_putw (s, pkt_afi);    /* AFI */
   stream_putc (s, pkt_safi);   /* SAFI */
-  if (afi == AFI_L2VPN)
-    nh_afi = AFI_L2VPN;
-  else if (nh_afi == AFI_MAX)
-    nh_afi = BGP_NEXTHOP_AFI_FROM_NHLEN(attr->extra->mp_nexthop_len);
+
+  /* Nexthop AFI */
+  if (peer_cap_enhe(peer, afi, safi)) {
+    nh_afi = AFI_IP6;
+  } else {
+    if (afi == AFI_L2VPN)
+      nh_afi = AFI_L2VPN;
+    else if (safi == SAFI_LABELED_UNICAST)
```

Comment:
If we do the same thing here for SAFI_MPLS_VPN then we fix the 4-nexthops problem from #572.

But I'm unsure if adding these exceptions here and there are the best way to go. Wouldn't it be better if we initialize `mp_nexthop_len` of vpn/labeled-unicast routes so BGP_NEXTHOP_AFI_FROM_NHLEN() actually works for all SAFIs?